ssp

```diff
@@ -8252,13 +8240,36 @@ sub check_for_cgls {
 ## END malware checks
 
 sub build_rpm_list {
-    return if @extended_rpm_list;    # already did this, don't do it again
+    my ($rpm_list) = @_;
+    return unless $rpm_list;
+    return map {
+        my ( $arch, $name, $version, $release ) = split( /\t/, $_ );
+        {
+            'name'    => defined $name    ? $name    : '',
+            'arch'    => defined $arch    ? $arch    : '',
+            'version' => defined $version ? $version : '',
+            'release' => defined $release ? $release : '',
+        }
+    } split( /\n/, $rpm_list );
```

Comment:
Dang the readability here is definitely a bit obfuscated but it does work.  One of those cases where tidy almost makes this line harder to understand.  It's good though!

Reply:
Most of this pull request was a cut/paste re-arrangement of some of the recent Grumpsquad additions to separate subs (with some new glue to use them), like this one, so it could be used in the forgotten DNSONLY and non-cPanel code paths. :-)

Ideally I'd like to merge these code paths together in the future and find a better way to decide which checks to run (or not) on different platforms so the less-used paths are less likely to suffer bit-rot.